### DISCUSSION TITLE
GSOC 2024

### DISCUSSION THREAD

**Discussion by krishbansal10** [sosy-lab/benchexec#1003] [General]
Hi @PhilippWendler,

I am really excited and looking forward to working on a project under your guidance, i.e. Improvements to generated tables for BenchExec. How am I supposed to start contributing and is Linux knowledge required?

Greetings,
Krish

**Comment by PhilippWendler** [comment]
Hi and welcome!

I don't think there is anything particular that you need to know before contributing. For getting to know BenchExec I would in general recommend to first get started with understanding BenchExec, for example by reading our [paper](https://doi.org/10.1007/s10009-017-0469-y) and trying out BenchExec in practice. The benchmarking part requires Linux, but if you want to work only on the HTML tables, you don't need it in principle. You should be able to run `table-generator` with some existing result files and `table-generator` only needs Python and works on Windows. There are some (old) existing result files in the directory `benchexec/tablegenerator/test_integration/` and [here](https://github.com/sosy-lab/benchexec/tree/gh-pages/example-table).

For the generated tables, we have an [online example](https://sosy-lab.github.io/benchexec/example-table/svcomp-simple-cbmc-cpachecker.table.html). There are many improvement ideas about them, have a look at our issues labeled with [HTML table](https://github.com/sosy-lab/benchexec/issues?q=is%3Aopen+label%3A%22HTML+table%22). Our developer instructions and the source code are in [this part of the repository](https://github.com/sosy-lab/benchexec/tree/main/benchexec/tablegenerator/react-table).

If you have questions just ask me.

Greetings
Philipp
